apps/dashboard/src/routes/+layout.server.ts

```diff
@@ -1,23 +1,33 @@
-import { supabaseAdminClient } from '$lib/supabase/supabase-admin-client';
-import { redirect, error } from '@sveltejs/kit';
 import type { LayoutServerLoad } from './$types';
+import { building } from '$app/environment';
+import { stripeAdminClient } from '$lib/stripe/stripe-admin-client';
+import { upsertPriceRecord, upsertProductRecord } from '$lib/stripe/stripe-billing-helpers';
 
-export const load: LayoutServerLoad = async ({ url, locals: { getSession } }) => {
-	const isDevelopment = process.env.NODE_ENV === 'development';
-	const link = isDevelopment ? url.searchParams.get('link') : url.toString();
+// TODO: Maybe we should move this somewhere else?
+if (!building) {
+	stripeAdminClient.products
+		.list()
+		.then(async (products) => {
+			for (const product of products.data) {
+				await upsertProductRecord(product);
+			}
+		})
+		.catch((e) => {
+			console.log(e);
+		});
 
-	if (link) {
-		const { data: linkData, error: linkError } = await supabaseAdminClient
-			.from('dynamic_links')
-			.select('deep_link')
-			.eq('link', link)
-			.single();
+	stripeAdminClient.prices
+		.list()
+		.then(async (prices) => {
+			for (const price of prices.data) {
+				await upsertPriceRecord(price);
+			}
+		})
+		.catch((e) => {
+			console.log(e);
+		});
+}
```

Comment:
This fetches the products and prices from stripe and throws it into the database. The basejump guys use a script in order to achieve this https://github.com/usebasejump/basejump/blob/main/scripts/sync-stripe.ts

I believe this part should only run once on startup
https://kit.svelte.dev/docs/building-your-app#during-the-build

Do you want to move this somewhere else?